dango/testing/tests/dex.rs

```diff
@@ -348,3 +348,127 @@ fn dex_works(
         .collect::<BTreeMap<_, _>>();
     assert_eq!(orders, remaining_orders);
 }
+
+#[test]
+fn cancel_order() {
+    let (mut suite, mut accounts, _, contracts) = setup_test_naive();
+
+    // Record the user's balance
+    suite.balances().record(accounts.user1.address());
+
+    // Add order to the order book
+    suite
+        .execute(
+            &mut accounts.user1,
+            contracts.dex,
+            &dex::ExecuteMsg::SubmitOrder {
+                base_denom: DANGO_DENOM.clone(),
+                quote_denom: USDC_DENOM.clone(),
+                direction: Direction::Bid,
+                amount: Uint128::new(100),
+                price: Udec128::new(1),
+            },
+            Coins::one(USDC_DENOM.clone(), 100).unwrap(),
+        )
+        .should_succeed();
+
+    // Cancel the order
+    suite
+        .execute(
+            &mut accounts.user1,
+            contracts.dex,
+            &dex::ExecuteMsg::CancelOrders {
+                order_ids: BTreeSet::from([!0]),
+            },
+            Coins::one(DANGO_DENOM.clone(), 1).unwrap(),
+        )
+        .should_succeed();
+
+    // Check that the user balance has not changed
+    assert_eq!(
+        suite
+            .balances()
+            .changes(accounts.user1.address())
+            .get(&USDC_DENOM)
+            .unwrap(),
+        &BalanceChange::Unchanged
+    );
+
+    // Check that order does not exist
+    let orders = suite
+        .query_wasm_smart(contracts.dex, QueryOrdersRequest {
+            start_after: None,
+            limit: None,
+        })
+        .unwrap();
+    assert!(orders.is_empty())
```

Comment:
Similarly, use the `grug::ResultExt` syntax:

```rust
suite
    .query_wasm_smart(contracts.dex, QueryOrdersRequest {
        start_after: None,
        limit: None,
    })
    .should(BTreeMap::is_empty);
```

Reply:
ack